### PR TITLE
 [jit] Add torch._C._nn built-in, more weak fns

### PR DESCRIPTION
**Pull request by driazati**
This PR adds functions defined in `torch._C._nn` as builtin functions (including inplace variants). This allows for the conversion of more functions to weak script

NB: many `torch.nn.functional` functions will have to be slightly rewritten to avoid early returns (as with `threshold` in this PR)

Converts these functions to weak script:
* `threshold`
* `relu`
* `hardtanh`
* `relu6`
* `elu`
* `selu`
* `celu`
* `leaky_relu`
* `rrelu`
* `tanh`
* `sigmoid`

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -708,15 +712,19 @@ def threshold(input, threshold, value, inplace=False):
...
+    # type: (Tensor, bool) -> Tensor
     r"""relu(input, inplace=False) -> Tensor
 
     Applies the rectified linear unit function element-wise. See
     :class:`~torch.nn.ReLU` for more details.
     """
     if inplace:
-        return torch.relu_(input)
-    return torch.relu(input)
+        result = torch.relu_(input)
```
Comment (zdevito) [inline torch/nn/functional.py]: We should record handling more complicated return statement configurations as a feature task at p2.

**Review by zdevito — APPROVED** [review]
